### DISCUSSION TITLE
graalvm native-image supported?

### DISCUSSION THREAD

**Discussion by lost22git** [activej/activej#162] [Q&A]

**Comment by eduard-vasinskyi** [comment]
Hi, @test210820 @angviza @richarddd

Yes, you can use Graal native-image tool to build executables out of ActiveJ code.

We have added a simple [tutorial](https://github.com/activej/activej/tree/master/examples/tutorials/native-image) on how to build an ActiveJ HTTP web server as a native image. Although the application itself is simple and contains only a few lines of code, it is actually a full-fledged web server that uses [ActiveJ Inject](https://activej.io/inject), [Launcher](https://activej.io/boot/launcher), [HTTP](https://activej.io/http), [Service graph](https://activej.io/boot/servicegraph), [Eventloop](https://activej.io/async-io/eventloop), and [Promises](https://activej.io/async-io/promise) modules. 

The resulting web server has a surprisingly fast startup speed (only a few milliseconds) and a small executable size (~4 MB after processing the executable with the UPX tool).

When you build a native image for an ActiveJ application that uses ActiveJ Inject annotations (`@Inject`, `@Provides`, etc.) you need to specify classes that use these annotations in a reflection configuration file ([reflectionconfig.json](https://github.com/activej/activej/blob/master/examples/tutorials/native-image/config/reflectionconfig.json) in the tutorial). Other than that there are no additional prerequisites for building ActiveJ native image applications.

If you need help with building a native image for the ActiveJ application feel free to ask. We will be glad to help you.

**Comment by richarddd** [comment]
I can confirm that this works. Please go ahead and close the issue :)
